pkg/consuldp/consul_dataplane.go

```diff
@@ -195,6 +186,26 @@ func (cdp *ConsulDataplane) Run(ctx context.Context) error {
 	return <-doneCh
 }
 
+func (cdp *ConsulDataplane) startDNSProxy(ctx context.Context) (dns.DNSServerInterface, error) {
+	dnsClientInterface := pbdns.NewDNSServiceClient(cdp.serverConn)
+
+	dnsServer, err := dns.NewDNSServer(dns.DNSServerParams{
+		BindAddr: cdp.cfg.DNSServer.BindAddr,
+		Port:     cdp.cfg.DNSServer.Port,
+		Client:   dnsClientInterface,
+		Logger:   cdp.logger,
+	})
+	if err == dns.ErrServerDisabled {
+		return nil, err
```

Comment:
```suggestion
		return nil, nil
```

If you do this then you wont have to check for ErrServerDisabled further up.